sycl/test-e2e/lit.cfg.py

```diff
@@ -541,6 +541,10 @@ def open_check_file(file_name):
     else:
         config.substitutions.append(("%hip_options", ""))
 
+# Add ROCM_PATH from system environment, this is used by clang to find ROCm
+# libraries in non-standard installation locations.
+llvm_config.with_system_environment("ROCM_PATH")
```

Comment:
Should we instead update the CMake code [here](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/CMakeLists.txt#L96)? When users pass '%hip_options' on Windows it also passes `--rocm-path` based on that CMake var [here](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/lit.cfg.py#L533), if needed we could make this happen for Linux too

Reply:
Actually I'm seeing now we should already check that envvar to set the cmake var, so maybe the problem is just that we don't pass `--rocm-path` on linux?

Reply:
This is needed for every single test not just when `%hip_options` are passed.

There's two different things going on, the `%hip_options` are used specifically for interop tests that use HIP code alongside SYCL code, they're used to link the runtime HIP libraries and include the HIP runtime headers.

But then for all SYCL tests, when compiling for AMD, `clang` uses some bitcode libraries provided by ROCm so it also needs to find the ROCm installation.

This patch is fixing that second use case, we could potentially use `--rocm-path` instead but that would have to be added to the regular `%{build}` substitution not just `%hip_options`, which may be slightly cleaner, but using `ROCM_PATH` is a bit easier and more flexible.

Reply:
got it, thanks. i didnt know the rocm path was needed when compiling for amd when not using hip.

i would prefer we minimize environment var usage in tests so i would prefer we remove the passing of `--rocm-path` as part of `hip_options` on Windows and just always add it to the `%{build}` substitution on all OSs when compiling for AMD

@aelovikov-intel Thoughts?

Reply:
IIUC, we can use this and remove all `--rocm-path`s. User's env is supposed to have `ROCM_PATH` (we copy it from there, after all), so they won't need to add extra env variables when reproducing outside LIT.

Reply:
yeah it seems like `ROCM_PATH` must always exist and point to the rocm install, i thought it was optional, in that case im fine to use the envvar

Reply:
I did have a look at the CMake and on top of that the HIP CMake is annoying so it's not straightforward to get the rocm path from it, so this is actually a lot easier.

I don't have a setup with Windows + ROCm, I can remove the `--rocm-path` there, but it's untested, and I'm not sure if `ROCM_PATH` works the same on Windows. That's the only place that uses `--rocm-path` as far as I can tell.

Reply:
let's try removing windows, if it fails just add it back, i vaguely remember the rocm detection on windows in upstream clang being basically non-existent so we might need it there